src/utils/FormattingUtils.ts

```diff
@@ -0,0 +1,165 @@
+import { ethers, BigNumber } from "ethers";
+import { createFormatFunction } from "@uma/common";
+import { DEFAULT_BLOCKCHAIN_EXPLORER_DOMAIN, PUBLIC_NETWORKS } from "../constants";
+
+export { createFormatFunction };
+
+export type BigNumberish = ethers.BigNumberish;
+export type BN = ethers.BigNumber;
+
+/**
+ * Parse a stringified number into a BigNumber.
+ * @param num The number to parse.
+ * @returns The parsed BigNumber.
+ */
+export function toWei(num: BigNumberish): BN {
+  return ethers.utils.parseEther(num.toString());
+}
+
+/**
+ * Parse a stringified number into a BigNumber with 9 decimal places.
+ * @param num The number to parse.
+ * @returns The parsed BigNumber.
+ */
+export function toGWei(num: BigNumberish): BN {
+  return ethers.utils.parseUnits(num.toString(), 9);
+}
+
+/**
+ * Converts a stringified number into a BigNumber. If the string version of the num contains a `.` then it is a number which needs to be parsed to a string int.
+ * @param num The number to parse.
+ * @returns The parsed BigNumber.
+ * @note This is a temporary function until we can backport support for decimal points to @across-protocol/sdk-v2.
+ */
+export const toBN = (num: BigNumberish): BN => {
+  // If the string version of the num contains a `.` then it is a number which needs to be parsed to a string int.
+  if (num.toString().includes(".")) {
+    return BigNumber.from(parseInt(num.toString()));
+  }
+  return BigNumber.from(num.toString());
+};
+
+export const formatFeePct = (relayerFeePct: BigNumber): string => {
+  // 1e18 = 100% so 1e16 = 1%.
+  return createFormatFunction(2, 4, false, 16)(toBN(relayerFeePct).toString());
+};
+
+/**
+ * Creates an etherscan link for a transaction or address on a given network.
+ * @param txHashOrAddress The transaction hash or address to link to.
+ * @param chainId The chainId to link to.
+ * @returns The etherscan link. This link will be formatted in markdown.
+ */
+export function etherscanLink(txHashOrAddress: string, chainId: number | string): string {
+  return _createEtherscanLinkMarkdown(txHashOrAddress, Number(chainId)) ?? "<>";
+}
+
+/**
+ * Resolves a domain to an etherscan link for a given network.
+ * @param networkId The network to link to.
+ * @returns The etherscan link. If the networkId is not supported, the default etherscan mainnet link will be returned.
+ */
+export function createEtherscanLinkFromTx(networkId: number): string {
+  return PUBLIC_NETWORKS[networkId]?.etherscan ?? DEFAULT_BLOCKCHAIN_EXPLORER_DOMAIN;
+}
+
+/**
+ * Shortens a lengthy hexadecimal string to a shorter version with an ellipsis in the middle.
+ * @param hex A hexadecimal string to be shortened.
+ * @returns The shortened hexadecimal string.
+ * @example shortenHexString("0x772871a444c6e4e9903d8533a5a13101b74037158123e6709470f0afbf6e7d94") -> "0x7787...7d94"
+ */
+export function createShortHexString(hex: string): string {
+  return hex.substring(0, 5) + "..." + hex.substring(hex.length - 6, hex.length);
```

Comment:
Consider allowing the width to be variable, as implemented here? https://github.com/across-protocol/relayer-v2/blob/e38356ed0496f85a44bca0461a34200a0139d388/scripts/spokepool.ts#L23

Reply:
I think we should change the function signature in a different PR.